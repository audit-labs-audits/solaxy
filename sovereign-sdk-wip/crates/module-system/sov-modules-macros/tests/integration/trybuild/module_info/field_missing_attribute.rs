use sov_modules_api::{Context, Module, ModuleId, ModuleInfo, Spec, StateMap, TxState};

#[derive(ModuleInfo)]
struct TestStruct<S: Spec> {
    #[id]
    pub id: ModuleId,

    test_state1: StateMap<u32, u32>,

    #[state]
    test_state2: StateMap<Vec<u8>, u64>,

    #[phantom]
    phantom: std::marker::PhantomData<S>,
}

impl<S: Spec> Module for TestStruct<S> {
    type Spec = S;

    type Config = ();

    type CallMessage = ();

    type Event = ();

    fn call(
        &mut self,
        _message: Self::CallMessage,
        _context: &Context<Self::Spec>,
        _state: &mut impl TxState<S>,
    ) -> Result<(), sov_modules_api::Error> {
        todo!()
    }
}

fn main() {}
