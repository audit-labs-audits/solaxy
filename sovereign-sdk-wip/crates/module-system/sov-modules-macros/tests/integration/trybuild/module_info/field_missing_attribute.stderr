error: The field `test_state1` is missing an attribute: add `#[kernel_module]`, `#[module]`, `#[state]`, `#[id]`, or #[phantom].
 --> tests/integration/trybuild/module_info/field_missing_attribute.rs:8:5
  |
8 |     test_state1: StateMap<u32, u32>,
  |     ^^^^^^^^^^^

error[E0277]: the trait bound `std::vec::Vec<u8>: FromStr` is not satisfied
  --> tests/integration/trybuild/module_info/field_missing_attribute.rs:11:18
   |
11 |     test_state2: StateMap<Vec<u8>, u64>,
   |                  ^^^^^^^^^^^^^^^^^^^^^^ the trait `FromStr` is not implemented for `std::vec::Vec<u8>`
   |
   = help: the following other types implement trait `FromStr`:
             AddressBech32
             Amount
             ColorChoice
             CredentialId
             HeaderName
             HeaderValue
             HexString<T>
             IpAddr
           and $N others
note: required by a bound in `sov_modules_api::containers::map::NamespacedStateMap`
  --> $WORKSPACE/crates/module-system/sov-modules-api/src/containers/map.rs
   |
   | pub struct NamespacedStateMap<N, K, V, Codec = BorshCodec>
   |            ------------------ required by a bound in this struct
...
   |     K: FromStr + std::fmt::Display,
   |        ^^^^^^^ required by this bound in `NamespacedStateMap`

error[E0277]: `std::vec::Vec<u8>` doesn't implement `std::fmt::Display`
  --> tests/integration/trybuild/module_info/field_missing_attribute.rs:11:18
   |
11 |     test_state2: StateMap<Vec<u8>, u64>,
   |                  ^^^^^^^^^^^^^^^^^^^^^^ `std::vec::Vec<u8>` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `std::vec::Vec<u8>`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
note: required by a bound in `sov_modules_api::containers::map::NamespacedStateMap`
  --> $WORKSPACE/crates/module-system/sov-modules-api/src/containers/map.rs
   |
   | pub struct NamespacedStateMap<N, K, V, Codec = BorshCodec>
   |            ------------------ required by a bound in this struct
...
   |     K: FromStr + std::fmt::Display,
   |                  ^^^^^^^^^^^^^^^^^ required by this bound in `NamespacedStateMap`
